packages/editor/src/store/resolvers.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Internal dependencies
+ */
+import {
+	resetAutosave,
+} from './actions';
+import { apiFetch } from './controls';
+
+/**
+ * Request autosave data from the REST API.
+ *
+ * @param {string} postId The Post to retrieve the autosave for.
+ */
+export function* getAutosave( postId ) {
+	const autosaveResponse = yield apiFetch( { path: `/wp/v2/posts/${ postId }/autosaves?context=edit` } );
+	if ( autosaveResponse && autosaveResponse[ 0 ] ) {
+		yield resetAutosave( autosaveResponse[ 0 ] );
```

Comment:
One issue to solve is that the autosave should only be stored when it's newer than the current post's modified date, so that it matches this bit of logic - https://github.com/WordPress/gutenberg/blob/master/lib/client-assets.php#L1276